tools/accuracy_checker/accuracy_checker/preprocessor/audio_preprocessing.py

```diff
@@ -178,9 +179,307 @@ def _parse_duration(self, duration):
 class NormalizeAudio(Preprocessor):
     __provider__ = 'audio_normalization'
 
+    @classmethod
+    def parameters(cls):
+        parameters = super().parameters()
+        parameters.update({
+            'int16mode': BoolField(optional=True, default=False, description="Normalization to int16 range"),
+        })
+        return parameters
+
+    def configure(self):
+
+        self.int16mode = self.get_value_from_config('int16mode')
+
     def process(self, image, annotation_meta=None):
         sound = image.data
-        sound = (sound - np.mean(sound)) / (np.std(sound) + 1e-15)
+        if self.int16mode:
+            sound = sound / np.float32(0x7fff)
+        else:
+            sound = (sound - np.mean(sound)) / (np.std(sound) + 1e-15)
+
         image.data = sound
 
         return image
+
+class HanningWindow(Preprocessor):
+    __provider__ = 'hanning_window'
+
+    @classmethod
+    def parameters(cls):
+        parameters = super().parameters()
+        parameters.update({
+            'base': NumberField(
+                optional=True,
+                default=512,
+                description="Window length"
+            ),
+        })
+        return parameters
+
+    def configure(self):
+
+        self.base = self.get_value_from_config('base')
+        self.window = np.hanning(self.base)
+
+
+    def process(self, image, annotation_meta=None):
+        image.data = image.data * self.window
+
+        return image
+
+class AudioSpectrogram(Preprocessor):
+    __provider__ = 'audio_spectrogram'
+
+    @classmethod
+    def parameters(cls):
+        parameters = super().parameters()
+        parameters.update({
+            'fftbase': NumberField(optional=True, default=512, description="Base of FFT, samples"),
+            'magnitude_squared': BoolField(optional=True, default=True, description="Square spectrum magnitudes"),
+            'skip_channels': BoolField(optional=True, default=False, description="Skips channels dimension"),
+        })
+        return parameters
+
+    def configure(self):
+
+        self.fftbase = self.get_value_from_config('fftbase')
+        self.magnutide_squared = self.get_value_from_config('magnitude_squared')
+        self.skip_channels = self.get_value_from_config('skip_channels')
+
+    def process(self, image, annotation_meta=None):
+        frames = image.data
+        if self.skip_channels:
+            frames = frames.squeeze()
+
+        pspec = np.absolute(np.fft.rfft(frames, self.fftbase))
```

Comment:
ignore pylint false positive
```suggestion
        pspec = np.absolute(np.fft.rfft(frames, self.fftbase)) # pylint:disable=W9904
```